lib/Dialect/OM/OMOps.cpp

```diff
@@ -475,6 +504,18 @@ void circt::om::ClassExternOp::replaceFieldTypes(AttrTypeReplacer replacer) {
   replaceClassLikeFieldTypes(*this, replacer);
 }
 
+//===----------------------------------------------------------------------===//
+// ClassFieldsOp
+//===----------------------------------------------------------------------===//
+//
+LogicalResult circt::om::ClassFieldsOp::verify() { 
+  auto fieldLocs = this->getFieldLocs();
+  if (fieldLocs.has_value())
+    if (fieldLocs.value().size() != this->getFields().size())
+      return this->emitOpError("size of field_locs does not match number of fields");
```

Comment:
It might be helpful to print out the size of each, or even printing out the actual fieldLocs attributes and the field names, or something like that. Up to you.